lib/completely/completions.rb

```diff
@@ -54,7 +54,7 @@ def tester
 
     def patterns!
       config.map do |text, completions|
-        Pattern.new text, completions
+        Pattern.new text, completions, pattern_function_name
```

Comment:
`Pattern` now receives a function name, which is the name of the helper filter function. It is used in `Pattern#compgen` to generate the `compgen` string which now calls a function rather than just using the string straight up.